examples/dynamic_mat_transpose.cpp

```diff
@@ -0,0 +1,427 @@
+          int row = by * TILE_DIM + ty;  // Matrix row index
+
+          // Bounds check
+          if (row < N_r && col < N_c) {
+            Atview(col, row) = Tile[ty][tx];
+          }
+        }
+      }
+
+    }
+  }
+  // _mattranspose_localarray_cstyle_end
+
+  checkResult<int>(Atview, N_c, N_r);
+  // printResult<int>(Atview, N_c, N_r);
+
+  //----------------------------------------------------------------------------//
+
+  std::cout << "\n Running RAJA matrix transpose w/ dynamic shared memory ...\n";
+
+#if defined(RAJA_ENABLE_HIP)
+
+  //Hip requires device side pointers
+  int *d_A = nullptr, *d_At = nullptr;
+
+  if(select_cpu_or_gpu == RAJA::DEVICE) {
+    d_A =  memoryManager::allocate_gpu<int>(N_r * N_c);
+    d_At = memoryManager::allocate_gpu<int>(N_r * N_c);
+
+    hipErrchk(hipMemcpy( d_A, A, N_r * N_c * sizeof(int), hipMemcpyHostToDevice ));
+
+    //switch host/device pointers so we can reuse the views
+    switch_ptrs(d_A, A);
+    switch_ptrs(d_At, At);
+  }
+#endif
+
+
+  constexpr size_t dynamic_shared_mem = TILE_DIM * TILE_DIM * sizeof(int);
+
+  RAJA::launch<launch_policy>(select_cpu_or_gpu, dynamic_shared_mem,
+    RAJA::Grid(RAJA::Teams(outer_Dimr, outer_Dimc),
+                     RAJA::Threads(TILE_DIM, TILE_DIM)),
+       [=] RAJA_HOST_DEVICE(RAJA::LaunchContext ctx)
+  {
+
+    RAJA::loop<outer1>(ctx, RAJA::RangeSegment(0, outer_Dimr), [&] (int by){
+        RAJA::loop<outer0>(ctx, RAJA::RangeSegment(0, outer_Dimc), [&] (int bx){
+
+            //specify view properties
+            using shmem_type = int; constexpr int tile_dim = 2;
+            const size_t shmem_size = TILE_DIM*TILE_DIM;
+
+            //Returns a RAJA view for simplified indexing
+	    auto Tile_1 = ctx.getSharedMemoryView<shmem_type, tile_dim>(shmem_size, TILE_DIM, TILE_DIM);
+
+            RAJA::loop<inner1>(ctx, RAJA::RangeSegment(0, TILE_DIM), [&] (int ty){
+              RAJA::loop<inner0>(ctx, RAJA::RangeSegment(0, TILE_DIM), [&] (int tx){
+
+                  int col = bx * TILE_DIM + tx;  // Matrix column index
+                  int row = by * TILE_DIM + ty;  // Matrix row index
+
+                  // Bounds check
+                  if (row < N_r && col < N_c) {
+                    Tile_1(ty,tx) = Aview(row, col);
+                  }
+
+                });
+              });
+
+            //need a barrier
+            ctx.teamSync();
+
+            RAJA::loop<inner1>(ctx, RAJA::RangeSegment(0, TILE_DIM), [&] (int ty){
+              RAJA::loop<inner0>(ctx, RAJA::RangeSegment(0, TILE_DIM), [&] (int tx){
+
+                  int col = bx * TILE_DIM + tx;  // Matrix column index
+                  int row = by * TILE_DIM + ty;  // Matrix row index
+
+                  // Bounds check
+                  if (row < N_r && col < N_c) {
+                    Atview(col, row) = Tile_1(ty, tx);
+                  }
+
+                });
+              });
+
+            ctx.releaseSharedMemory();
```

Comment:
What does this do? Is the memory not allocated before we enter the body of the launch?

Reply:
Oh, I see its required because of the the offsetting. Indeed there is no way to auto reset that given how it works.